src/app/community/community.component.ts

```diff
@@ -60,27 +61,36 @@ export class CommunityComponent implements OnInit, AfterViewInit {
       }, (error) => this.message = 'There was a problem getting Communities');
   }
 
-  deleteClick(community) {
-    this.deleteDialog = this.dialog.open(DialogsDeleteComponent, {
+  updateClick(community, change) {
+    this.editDialog = this.dialog.open(DialogsEditComponent, {
       data: {
-        okClick: this.deleteCommunity(community),
+        okClick: this.updateCommunity(community, change),
+        changeType: change,
         type: 'community',
         displayName: community.name
       }
     });
   }
 
-  deleteCommunity(community) {
+  updateCommunity(community, change) {
     // Return a function with community on its scope to pass to delete dialog
     return () => {
     // With object destructuring colon means different variable name assigned, i.e. 'id' rather than '_id'
       const { _id: id, _rev: rev } = community;
-      this.couchService.delete('communityregistrationrequests/' + id + '?rev=' + rev)
+      if (change !== 'delete') {
+        community['registrationRequest'] = change;
+        this.couchService.put('communityregistrationrequests/' + id + '?rev=' + rev, community)
```

Comment:
It would improve readability to have an `updateCommunity` and `deleteCommunity` function that is returned by this function based on the change type rather than what you have here.